pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TransactionsBase.java

```diff
@@ -432,4 +433,24 @@ protected void validateTopicName(String property, String namespace, String encod
             throw new RestException(Response.Status.PRECONDITION_FAILED, "Topic name is not valid");
         }
     }
+
+    protected CompletableFuture<Void> internalScaleTransactionCoordinators(int replicas) {
+        CompletableFuture<Void> completableFuture = new CompletableFuture<>();
```

Comment:
We don't need to create the CompletableFuture here but complete it in another thread.